test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -8,43 +8,63 @@
+  @Mock
+  private final GameData gameData = Mockito.mock(GameData.class);
 
-  final int unitCountDefaultPlayerUnitTypeOne = getUnitsOfDefaultPlayerOfUnitTypeOne().size();
-  final int unitCountDefaultPlayerUnitTypeTwo = getUnitsOfDefaultPlayerOfUnitTypeTwo().size();
-  final int unitCountDefaultPlayer = unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo;
-  final Unit unitDefaultPlayer1 = new Unit(unitTypeOne, defaultPlayerID, gameData);
-  final Unit unitDefaultPlayer2 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
-  final Unit unitDefaultPlayer3 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
+  @Before
+  public void setup() {
+    final UnitTypeList unitTypeList = new UnitTypeList(gameData);
+    unitTypeList.addUnitType(unitTypeOne);
+    unitTypeList.addUnitType(unitTypeTwo);
+    Mockito.when(gameData.getUnitTypeList()).thenReturn(unitTypeList);
+  }
+
+  final private UnitType unitTypeOne = new UnitType("Unit Type 1", gameData);
+  final private UnitType unitTypeTwo = new UnitType("Unit Type 2", gameData);
+  @Mock
+  final private MyPlayerID defaultPlayerID = new MyPlayerID();
+  final private PlayerID otherPlayerID = PlayerID.NULL_PLAYERID;
+  final private UnitCollection unitCollection = new UnitCollection(defaultPlayerID, gameData);
+
+  final private int unitCountDefaultPlayerUnitTypeOne = getUnitsOfDefaultPlayerOfUnitTypeOne().size();
 
   private Collection<Unit> getUnitsOfDefaultPlayerOfUnitTypeOne() {
     final Collection<Unit> units = new ArrayList<Unit>();
     units.add(unitDefaultPlayer1);
     return units;
   }
 
+  final private int unitCountDefaultPlayerUnitTypeTwo = getUnitsOfDefaultPlayerOfUnitTypeTwo().size();
+
   private Collection<Unit> getUnitsOfDefaultPlayerOfUnitTypeTwo() {
     final Collection<Unit> units = new ArrayList<Unit>();
     units.add(unitDefaultPlayer2);
     units.add(unitDefaultPlayer3);
     return units;
   }
 
-  final int unitCountOtherPlayerUnitTypeOne = 3;
-  final int unitCountOtherPlayerUnitTypeTwo = 4;
+  final private int unitCountDefaultPlayer = unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo;
+  final private Unit unitDefaultPlayer1 = new Unit(unitTypeOne, defaultPlayerID, gameData);
+  final private Unit unitDefaultPlayer2 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
+  final private Unit unitDefaultPlayer3 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
+
+  final private int unitCountOtherPlayerUnitTypeOne = 3;
+  final private int unitCountOtherPlayerUnitTypeTwo = 4;
   final int unitCountOtherPlayer = unitCountOtherPlayerUnitTypeOne + unitCountOtherPlayerUnitTypeTwo;
+
+  final int unitCountUnitTypeOne = unitCountDefaultPlayerUnitTypeOne + unitCountOtherPlayerUnitTypeOne;
+  final int unitCountUnitTypeTwo = unitCountDefaultPlayerUnitTypeTwo + unitCountOtherPlayerUnitTypeTwo;
+
   final Unit unitOtherPlayer1 = new Unit(unitTypeOne, otherPlayerID, gameData);

```

Comment:
This is only used in the @Setup method. Inlining it in that method will make things simpler. No need for the `unitOtherPlayer1` through 3 variables.